Automattic/wp-calypso
# Onboarding: Show first post published modal

#### Changes proposed in this Pull Request

IMHO, the first post published modal is also one of the NUX feature, so I make it part of `wpcom-block-editor-nux` plugin.

* Create a `NuxModal` so that it can be re-used by `DraftPostModal` and `PostPublishedModal`
* Update the render function of `wpcom-block-editor-nux` plugin so that it can render both of `WelcomeTour` and `PostPublishedModal`. 
* Implement `PostPublishedModal` and it will do the following thing
  * Check the site has ever published any post before
  * If not and the current post status changed from non-publish to publish, the modal would be opened.
  * As we enable closing the modal when focusing outside, the `PostPublishedModal` would be close immediately. Thus, I try to delay open the `PostPublishedModal` to fix this problem.

![image](https://user-images.githubusercontent.com/13596067/137881002-ec7e996a-e687-43da-bbb6-81a074ee2ecd.png)

https://user-images.githubusercontent.com/13596067/137713967-29192c2e-eed8-41e0-8242-6db4f963d103.mov

#### Testing instructions



* Apply D68560-code to your sandbox
* Point your API to your sandbox
* Open another terminal and run
  * cd apps/wpcom-block-editor && yarn dev --sync
* Go to `/start?flags=signup/hero-flow`
* Finish the flow with following options
  * Select `Write` intent
  * Select `Choose a design` as a starting point (this step ensures running the headstart doesn't affect the first post published modal to show or not.
* After you publish the first post, you should see the first post published modal



Related to https://github.com/Automattic/wp-calypso/issues/56568
